src/file_system.rs

```diff
@@ -227,3 +232,31 @@ fn metadata() {
     );
     let _ = meta;
 }
+
+fn node_compatible_raw_canonicalize<P: AsRef<Path>>(path: P) -> io::Result<PathBuf> {
+    cfg_if! {
+        if #[cfg(windows)] {
+            // same logic with https://github.com/libuv/libuv/blob/d4ab6fbba4669935a6bc23645372dfe4ac29ab39/src/win/fs.c#L2774-L2784
+            let canonicalized = fs::canonicalize(path)?;
```

Comment:
I first thought about writing:
```rust
if let Some(stripped) = canonicalized.strip_prefix(UNC_PATH_PREFIX) {
}
```
but that didn't work, probably because `Path::strip_prefix` only supports stripping at a valid path component separation point.

Then, I thought about making substring from `canonicalized.as_os_str()` but that required unstable features (https://github.com/rust-lang/rust/issues/118485), so I avoided that.

Reply:
Applied some optimization in https://github.com/sapphi-red/oxc-resolver/pull/1